etc/lfcd.nu

```diff
@@ -0,0 +1,36 @@
+# Change working dir in shell to last dir in lf on exit (adapted from ranger).
+#
+# You need to add this to your Nushell Enviroment Config File
+# (Execute 'config env' in the nushell to open it).
+
+# You may also like to assign a key (Ctrl-O) to this command:
+# See the documentation: https://www.nushell.sh/book/line_editor.html#keybindings
+#
+# keybindings: [
+#    {
+#      name: lfcd
+#      modifier: control
+#      keycode: char_o
+#      mode: [emacs , vi_normal, vi_insert]
+#      event: {
+#        send: executehostcommand
+#        cmd: "lfcd"
+#      }
+#    }
+#  ]
+
+
+def-env lfcd [] {
```

Comment:
I was playing around with this function for a bit, I think it's better to use more native features than rather just directly translate directly from Bash. Could you please try the following to see if it works?

```nushell
def-env lfcd [] {
  let tmp = (mktemp)
  lf -last-dir-path $tmp
  if ($tmp | path exists) {
    let dir = (open --raw $tmp)
    rm -f $tmp
    if ($dir | path exists) and ($dir != $env.PWD) {
      cd $dir
    }
  }
}
````

Reply:
I can't think of a better way myself, but I think it's fine to keep the nested try/catch blocks. The important thing is to keep the code clean and readable, so that someone else is able to improve the code later on.

I'm not sure how much more value is provided by the `e` error object, if it doesn't add anything helpful (for me it just says `Directory not found`), you can exclude it. Also I would use `print -e` instead of `print` since that writes to stderr.

Reply:
OK I didn't realize that the error message was different for things like permission, in that case I think it's fine to keep the error then. Thanks!